ci/infra/libvirt/master-instance.tf

```diff
@@ -134,11 +134,17 @@ resource "null_resource" "master_reboot" {
     }
 
     command = <<EOT
-ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null $user@$host sudo reboot || :
-# wait for ssh ready after reboot
-until nc -zv $host 22; do sleep 5; done
-ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -oConnectionAttempts=60 $user@$host /usr/bin/true
+export sshopts="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -oConnectionAttempts=60"
+if ! ssh $sshopts $user@$host 'test -d /lib/modules/$(uname -r)'; then
```

Comment:
shouldn't we be checking `/var/run/reboot-needed` instead? or we dont have that installed on our images by default somehow (its part of core zypper I think?)?

Reply:
well, the idea was that we identify that we need a reboot (kernel modules installed do not match running kernel) and check that reboot happened (because running kernel matches installed modules). 

if we always want to reboot we're back to the original problem that it is hard to find out when reboot happened. 

I agree that touching /var/run/reboot-needed and waiting for the file to disappear could be a solution for that. I'll try that.

Reply:
actually this does not solve it, because I can't distinguish between unreachable node (which is an error) and 'node reachable but file no longer there (so reboot happened - success). 

I'm still thinking my current patch is the best one :/

Reply:
Found a way to use needs-rebooting (which is a zypper plugin).